spec/controllers/petitions_controller_spec.rb

```diff
@@ -272,6 +272,24 @@
         expect(response).to redirect_to("https://petition.parliament.uk/")
       end
     end
+
+    context "when the IP address is blocked" do
+      let(:petition) { Petition.find_by_action("Save the planet") }
+      let(:rate_limit) { RateLimit.first! }
+
+      before do
+        rate_limit.update!(blocked_ips: "0.0.0.0")
+      end
+
+      it "should not create a new petition and a signature" do
+        perform_enqueued_jobs do
+          post :create, params: { stage: "replay_email", petition_creator: params }
+        end
+
+        expect(petition).to be_nil
```

Comment:
I'd put a `expect {}.not_to change { Petition.count }` here, the `"Save the planet"` could fail to match for other reasons. I  was also expecting a check on signature since it's on the description

Reply:
I mean `find_by_action("Save the planet")` could fail to match